Tests/AmplitudeTests/Utilities/IdentifyInterceptorTests.swift

```diff
@@ -259,16 +219,30 @@ final class IdentifyInterceptorTests: XCTestCase {
         XCTAssertNotNil(events[0].userProperties)
         XCTAssertTrue(getDictionary(events[0].userProperties!).isEqual(to: ["$set": ["key-1": "value-1-1", "key-2": "value-2"]]))
 
-        // standard event should be modified to include intercepted user properties, flattened
+        // second $set only event should be intercepted
         let e2 = interceptor.intercept(event: testEvent2)
-        XCTAssertNotNil(e2)
-        XCTAssertEqual(e2!.eventType, "someEvent")
-        XCTAssertNotNil(e2!.userProperties)
-        XCTAssertTrue(getDictionary(e2!.userProperties!).isEqual(to: ["key-1": "value-1-2", "key-2": "value-2", "key-3": "value-3"]))
+        events = identifyStorage.events()
+        XCTAssertNil(e2)
+        XCTAssertEqual(events.count, 2)
+        XCTAssertNotNil(events[0].userProperties)
+        XCTAssertTrue(getDictionary(events[0].userProperties!).isEqual(to: ["$set": ["key-1": "value-1-1", "key-2": "value-2"]]))
+        XCTAssertNotNil(events[1].userProperties)
+        XCTAssertTrue(getDictionary(events[1].userProperties!).isEqual(to: ["$set": ["key-1": "value-1-2", "key-3": "value-3-1"]]))
+
+        // standard event should not be intercepted
+        let e3 = interceptor.intercept(event: testEvent3)
+        XCTAssertNotNil(e3)
+        XCTAssertEqual(e3!.eventType, "someEvent")
+        XCTAssertNotNil(e3!.userProperties)
+        XCTAssertTrue(getDictionary(e3!.userProperties!).isEqual(to: ["key-1": "value-1-3", "key-3": "value-3-2"]))
+
+        // intercept should be transferred to event storage
         events = identifyStorage.events()
         XCTAssertEqual(events.count, 0)
         events = storage.events()
-        XCTAssertEqual(events.count, 0)
+        XCTAssertEqual(events.count, 1)
+        XCTAssertNotNil(events[0].userProperties)
+        XCTAssertTrue(getDictionary(events[0].userProperties!).isEqual(to: ["$set": ["key-1": "value-1-2", "key-2": "value-2", "key-3": "value-3-1"]]))
```

Comment:
Having a hard time understanding this part of the test, might be due to my unfamiliarity with swift. How does this test that a single identify event is sent and then a single event is sent? (The first 2 sets should be collapsed into 1 identify with just set, then the event is fired)

Reply:
@ericyang-amplitude the `identifyStorage` is for the intercepted identifies. We don't collapse the values until we do the transfer to `storage`. So the 2 intercepts above become a single collapsed identify in `storage`.

Reply:
Copying my comment from slack for posterity's sake

does `storage` only hold identifies? I can't tell if the event being tested on line 243 is an identify or `someEvent `with the identify sets baked into it (which we said we wanted to avoid)

Reply:
@ericyang-amplitude 
I've added more checks (including `eventType == '$identify'` check)